designs/placement-groups.md

```diff
@@ -0,0 +1,65 @@
+# Placement Group Configuration
+
+## Goals
```

Comment:
I think we should write this design doc in a way that discusses the use-cases that we are trying to address through adding support for placement groups to the EC2NodeClass. From that, we can derive what the best API looks like here. 

Without the use-cases here, how can we say that users don't want to bring their own placement groups and want Karpenter to fully manage them?